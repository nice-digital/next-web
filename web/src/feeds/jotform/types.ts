export type FormID = `${number}`;

export type Duration = `${number}ms`;

export type NotFoundResponse = {
	responseCode: 404;
	message: string;
	content: "";
	duration: Duration;
	info: string;
};

export type NotAuthorizedResponse = {
	responseCode: 401;
	message: string;
	content: "";
	duration: Duration;
	info: string;
};

export type FormStatus = "ENABLED" | "DISABLED";

export type FormType = "LEGACY" | "CARD";

export type FormSuccessResponse = {
	responseCode: 200;
	message: "success";
	content: {
		id: string;
		username: string;
		title: string;
		height: number;
		status: FormStatus;
		/** Format like "2023-03-21 09:27:18" */
		created_at: string;
		updated_at: string;
		last_submission: string;
		new: `${number}`;
		count: `${number}`;
		type: FormType;
		favorite: `${number}`;
		archived: `${number}`;
		url: string;
	};
	duration: Duration;
};

export type GetFormResponse =
	| NotAuthorizedResponse
	| NotFoundResponse
	| FormSuccessResponse;
