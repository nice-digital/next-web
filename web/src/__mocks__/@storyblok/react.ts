export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
