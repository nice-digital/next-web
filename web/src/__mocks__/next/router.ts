export const useRouter = jest.fn().mockImplementation(() => {
	throw new Error("Provide a mock of useRouter for testing purposes");
});
