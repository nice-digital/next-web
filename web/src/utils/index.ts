import slugify from "@sindresorhus/slugify";
import dayjs from "dayjs";

import { Project, ProjectStatus } from "@/feeds/inDev/types";
import {
	ProductGroup,
	ProductLite,
	ProductTypeAcronym,
} from "@/feeds/publications/types";

import { dateFormat, dateFormatShort } from "./constants";

/**
 * Returns a a sortable date with just year, month and date (without time).
 * We never expose times to users so it makes sense to just sort by date instead.
 *
 * @param isoDateStr The full date string with timezone e.g. `2021-05-21T10:54:52.4655487`
 * @returns The sortable date, without the timezone e.g. `2021-05-21`
 */
export const stripTime = (isoDateStr: string): string =>
	isoDateStr.substring(0, isoDateStr.indexOf("T"));

/**
 * Formats the given ISO date string into a standard NICE string format
 *
 * @param isoDateStr the ISO date string for format
 * @param short whether to format the date in the standard short format
 * @returns The formatted date
 */
export const formatDateStr = (isoDateStr: string, short = false): string =>
	dayjs(isoDateStr).format(short ? dateFormatShort : dateFormat);

/**
 * Gets the path, relative to the root, of an indevelopment project overview page.
 *
 * Note: advice projects don't have project pages, so return null.
 *
 * @param project The indev project for which to get the URL
 * @returns The path of the project, relative to the root
 */
export const getProjectPath = (project: Project): string | null =>
	project.projectGroup == ProductGroup.Advice
		? null
		: project.status === ProjectStatus.Proposed
		? `/guidance/awaiting-development/${project.reference.toLowerCase()}`
		: `/guidance/indevelopment/${project.reference.toLowerCase()}`;

/**
 * Gets the path, relative to the root, of an published product overview page.
 *
 * @param product The product for which to get the URL
 * @returns The path of the product, relative to the root
 */
export const getProductPath = (
	product: Pick<ProductLite, "productGroup" | "id" | "productType" | "title">
): string => {
	let rootPath: string,
		productSlug = product.id.toLowerCase();

	switch (product.productGroup) {
		case ProductGroup.Guideline:
		case ProductGroup.Guidance:
		case ProductGroup.Standard:
			rootPath = "guidance";
			break;
		case ProductGroup.Advice:
			rootPath = "advice";
			break;
		case ProductGroup.Corporate:
			// There are 2 types of corporate products that have different URLs: corporate vs process and methods
			rootPath =
				product.productType === ProductTypeAcronym.ECD
					? "corporate"
					: "process";
			break;
		case ProductGroup.Other:
			if (product.productType !== ProductTypeAcronym.IND)
				throw Error(
					`Unsupported 'other' product type of ${product.productType}`
				);
			rootPath = "indicators";
			productSlug += `-${slugify(product.title)}`;
			break;
		default:
			throw `Unsupported product group ${product.productGroup} ${JSON.stringify(
				product
			)}`;
	}

	return `/${rootPath}/${productSlug}`;
};
