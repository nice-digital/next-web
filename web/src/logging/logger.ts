// import { useRouter } from "next/router";
import { useRouter } from "next/compat/router";
import pino, { Logger } from "pino";

import { niceLoggingPinoOptions } from "./nice-logging";

export const logger = pino(niceLoggingPinoOptions);

/**
 * React hook for using a logger within a React component (or other hook) that logs extra info to do with the current route.
 *
 * @returns A child pino logger that automatically logs extra routing info
 */
export const useLogger = (): Logger => {
	const router = useRouter();

	const pathname = router?.pathname;
	const query = router?.query;
	// 'Enrich' logs with extra request info like we do here in Guidance Web:
	// https://github.com/nice-digital/guidance-web/blob/master/Guidance.Web/RequestEnricher.cs#L19-L26
	return logger.child({
		Properties: {
			RequestPath: pathname,
			RequestQueryString: JSON.stringify(query), // Types have to match in ELK so make sure this is a string and not an object
			RequestUserAgent:
				typeof window === "undefined" ? undefined : window.navigator.userAgent,
		},
	});
};
