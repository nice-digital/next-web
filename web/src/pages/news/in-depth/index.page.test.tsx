import { ParsedUrlQuery } from "querystring";

import { getStoryblokApi, ISbStoryData } from "@storyblok/react";
import { render, screen, waitFor, within } from "@testing-library/react";
import { useRouter } from "next/router";
import { GetServerSidePropsContext } from "next/types";

import { logger } from "@/logger";
import MockStoryblokSuccessResponse from "@/test-utils/storyblok-news-articles-listing.json";
import MockServerErrorResponse from "@/test-utils/storyblok-server-error-response.json";
import { NewsStory } from "@/types/News";
import * as storyblokUtils from "@/utils/storyblok";
import { GENERIC_ERROR_MESSAGE } from "@/utils/storyblok";

import {
	getServerSideProps,
	InDepthArticleIndexPage,
	InDepthArticleProps,
} from "./index.page";

const mockStories = MockStoryblokSuccessResponse.data.stories;

describe("/news/in-depth/index.page", () => {
	(useRouter as jest.Mock).mockReturnValue({
		route: "/news/in-depth",
		pathname: "/news/in-depth",
		query: { page: "1" },
		asPath: "/news/in-depth?page=1",
		events: {
			on: jest.fn(),
			off: jest.fn(),
			emit: jest.fn(),
		},
		push: jest.fn(),
	});

	const mockConfig = {
		currentPage: 1,
		resultsPerPage: 6,
		startsWith: "news/in-depth/",
		query: {
			upperOutOfBoundPagination: "30",
			lowerOutOfBoundPagination: "-1",
			pageAt1: "1",
		},
		totalResults: 8,
	};

	const mockProps: InDepthArticleProps = {
		stories: mockStories as unknown as ISbStoryData<NewsStory>[],
		currentPage: mockConfig.currentPage,
		total: mockConfig.totalResults,
		perPage: mockConfig.resultsPerPage,
	};

	describe("InDepthArticleIndexPage", () => {
		it("should match snapshot for main content", () => {
			render(<InDepthArticleIndexPage {...mockProps} />);
			expect(document.body).toMatchSnapshot();
		});

		it("should render error page if validateRouteParams returns error", async () => {
			const errorMessage = "An error returned from getserversideprops";
			const { asFragment } = render(
				<InDepthArticleIndexPage error={errorMessage} />
			);

			expect(screen.getByText(errorMessage)).toBeInTheDocument();
			expect(asFragment()).toMatchSnapshot();
		});

		it("should render a page header", async () => {
			render(<InDepthArticleIndexPage {...mockProps} />);

			expect(
				screen.getByRole("heading", { level: 1, name: "In-depth" })
			).toBeInTheDocument();
		});

		it("should render breadcrumbs", () => {
			render(<InDepthArticleIndexPage {...mockProps} />);
			const navElement = screen.getByRole("navigation", {
				name: "Breadcrumbs",
			});
			const breadcrumbLinks = within(navElement).getAllByRole("link");
			expect(navElement).toBeInTheDocument();
			expect(breadcrumbLinks.length).toBe(2);
		});

		it("should render no content message if no stories are returned", () => {
			const mockPropsNoStories = {
				...mockProps,
				stories: [],
			};

			render(<InDepthArticleIndexPage {...mockPropsNoStories} />);

			expect(
				screen.getByText("Sorry there are no news articles available")
			).toBeInTheDocument();
		});

		it("should render a featured story if one is returned", () => {
			const mockPropsWithFeaturedStory = {
				...mockProps,
				featuredStory: mockProps.stories[0],
				stories: mockProps.stories.slice(1),
			};
			mockPropsWithFeaturedStory.featuredStory.name = "Featured story";

			render(<InDepthArticleIndexPage {...mockPropsWithFeaturedStory} />);

			expect(screen.getByText("Featured story")).toBeInTheDocument();
		});

		it("should render news navigation", () => {
			render(<InDepthArticleIndexPage {...mockProps} />);
			const navElement = screen.getByRole("navigation", {
				name: "News section navigation",
			});
			const newsNaviagtionLinks = within(navElement).getAllByRole("link");
			expect(navElement).toBeInTheDocument();
			expect(newsNaviagtionLinks.length).toBe(5);
		});

		it("should render an action banner for newsletters and alerts", () => {
			render(<InDepthArticleIndexPage {...mockProps} />);
			expect(
				screen.getByText("Sign up for our newsletters and alerts")
			).toBeInTheDocument();
		});

		it("should render a news list pagination component", () => {
			const { container } = render(<InDepthArticleIndexPage {...mockProps} />);
			const totalPages = Math.ceil(
				mockConfig.totalResults / mockConfig.resultsPerPage
			);
			expect(container.textContent).toHaveTextContentIgnoreTags(
				`Page ${mockConfig.currentPage} of ${totalPages}`
			);
		});

		describe("Accessibility features", () => {
			it("should render a hidden focusable heading for screen readers", () => {
				render(<InDepthArticleIndexPage {...mockProps} />);
				const hiddenFocusableHeading = screen.getByRole("heading", {
					level: 2,
					name: "In-depth articles list",
				});
				expect(hiddenFocusableHeading).toBeInTheDocument();
				expect(hiddenFocusableHeading).toHaveClass("visually-hidden");
			});
			it("should have an announcer for screen readers", async () => {
				const announcer = document.createElement("div");
				announcer.id = "__next-route-announcer__";
				announcer.setAttribute("aria-live", "assertive");
				announcer.setAttribute("role", "alert");
				document.body.appendChild(announcer);
				render(<InDepthArticleIndexPage {...mockProps} />);

				await waitFor(() =>
					expect(screen.getByRole("alert")).toHaveTextContent(
						"In-depth article listing page, 1 of 2"
					)
				);

				announcer.remove();
			});
		});
	});

	describe("getServerSideProps", () => {
		let validateRouteParamsSpy: jest.SpyInstance;
		beforeEach(() => {
			jest.mock("@/utils/storyblok", () => ({
				getStoryVersionFromQuery: jest.fn().mockReturnValue("published"),
			}));

			validateRouteParamsSpy = jest.spyOn(
				storyblokUtils,
				"validateRouteParams"
			);

			getStoryblokApi().get = jest
				.fn()
				.mockResolvedValue(MockStoryblokSuccessResponse);

			jest.useFakeTimers();
			jest.setSystemTime(new Date("2024-04-08"));
		});

		afterEach(() => {
			jest.clearAllMocks();
		});

		it("should return notFound when page is less than 1", async () => {
			const result = await getServerSideProps({
				query: { page: mockConfig.query.lowerOutOfBoundPagination },
				resolvedUrl: `${mockConfig.startsWith}?page=${mockConfig.query.lowerOutOfBoundPagination}`,
			} as unknown as GetServerSidePropsContext<ParsedUrlQuery>);

			expect(result).toEqual({
				notFound: true,
			});
		});

		it("should return notFound if pagination page is out of range", async () => {
			const result = await getServerSideProps({
				query: { page: mockConfig.query.upperOutOfBoundPagination },
				resolvedUrl: `${mockConfig.startsWith}?page=${mockConfig.query.upperOutOfBoundPagination}`,
			} as unknown as GetServerSidePropsContext<ParsedUrlQuery>);

			expect(result).toEqual({
				notFound: true,
			});
		});

		it("should call validateRouteParams with correct params", async () => {
			await getServerSideProps({
				query: { page: mockConfig.query.upperOutOfBoundPagination },
			} as unknown as GetServerSidePropsContext<ParsedUrlQuery>);

			expect(validateRouteParamsSpy).toHaveBeenCalledWith({
				query: { page: mockConfig.query.upperOutOfBoundPagination },
				sbParams: {
					starts_with: mockConfig.startsWith,
					per_page: mockConfig.resultsPerPage,
				},
			});
		});

		it("should return notFound if validateRouteParams returns notFound", async () => {
			validateRouteParamsSpy.mockResolvedValue({ notFound: true });

			const result = await getServerSideProps({
				query: { page: mockConfig.query.upperOutOfBoundPagination },
			} as unknown as GetServerSidePropsContext<ParsedUrlQuery>);

			expect(result).toEqual({ notFound: true });
		});

		it("should return redirect if validateRouteParams returns redirect", async () => {
			const mockRedirect = {
				redirect: { destination: "/news/articles", permanent: false },
			};

			validateRouteParamsSpy.mockResolvedValue(mockRedirect);

			const result = await getServerSideProps({
				query: { page: mockConfig.query.upperOutOfBoundPagination },
			} as unknown as GetServerSidePropsContext<ParsedUrlQuery>);

			expect(result).toEqual(mockRedirect);
		});

		it("should log and error if validateRouteParams throws an error", async () => {
			expect(jest.isMockFunction(logger.error)).toBe(true);
			const mockErrorMessage =
				"There was an error fetching this story. Please try again later.";
			const mockCatchError = new Error(mockErrorMessage, {
				cause: MockServerErrorResponse,
			});

			validateRouteParamsSpy.mockRejectedValue(mockCatchError);

			const mockContext = {
				query: { page: mockConfig.query.upperOutOfBoundPagination },
				req: { headers: {} },
			} as unknown as GetServerSidePropsContext<ParsedUrlQuery>;

			await getServerSideProps(mockContext);

			expect(logger.error).toHaveBeenCalled();
			expect(logger.error).toHaveBeenCalledWith(
				{
					errorCause: mockCatchError.cause,
					requestHeaders: mockContext.req.headers,
				},
				`Error fetching in-depth article listing at page ${mockConfig.query.upperOutOfBoundPagination} from gssp`
			);
		});

		it("should return error if validateRouteParams returns error", async () => {
			const mockErrorMessage =
				"<<<< There was an error fetching stories. Please try again later. >>>>> ";
			const mockCatchError = new Error(mockErrorMessage);

			validateRouteParamsSpy.mockRejectedValue(mockCatchError);

			const mockContext = {
				req: { headers: {} },
				params: {},
				query: { page: mockConfig.query.upperOutOfBoundPagination },
			} as unknown as GetServerSidePropsContext;

			const result = await getServerSideProps(mockContext);

			expect(result).toEqual({ props: { error: GENERIC_ERROR_MESSAGE } });
		});

		it("should return a general error message if there is a throw that is not an Error object", async () => {
			const mockCatchError = "some string error";

			validateRouteParamsSpy.mockRejectedValue(mockCatchError);

			const mockContext = {
				req: { headers: {} },
				params: {},
				query: { page: mockConfig.query.upperOutOfBoundPagination },
			} as unknown as GetServerSidePropsContext;

			const result = await getServerSideProps(mockContext);

			expect(result).toEqual({
				props: {
					error: GENERIC_ERROR_MESSAGE,
				},
			});
		});

		it("should return a correct props object if validateRouteParams returns success", async () => {
			const mockValidatedRouteParamsSuccessResponse = {
				featuredStory: mockStories[0],
				stories: mockStories.slice(1),
				total: MockStoryblokSuccessResponse.total,
				currentPage: mockConfig.currentPage,
				perPage: MockStoryblokSuccessResponse.perPage,
			};
			// MockStoryblokSuccessResponse
			validateRouteParamsSpy.mockResolvedValue(
				mockValidatedRouteParamsSuccessResponse
			);

			const result = await getServerSideProps({
				query: { page: mockConfig.query.pageAt1 },
			} as unknown as GetServerSidePropsContext<ParsedUrlQuery>);

			expect(result).toEqual({
				props: {
					featuredStory: MockStoryblokSuccessResponse.data.stories[0],
					stories: MockStoryblokSuccessResponse.data.stories.slice(1),
					perPage: MockStoryblokSuccessResponse.perPage,
					total: MockStoryblokSuccessResponse.total,
					currentPage: 1,
				},
			});
		});
	});
});
