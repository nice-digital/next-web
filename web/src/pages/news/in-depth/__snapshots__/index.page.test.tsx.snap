// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/news/in-depth/index.page InDepthArticleIndexPage should match snapshot for main content 1`] = `
<body>
  <div>
    <div
      class="page-header page-header--full-width-dark page-header--vertical-padding-loose"
      data-component="page-header"
      id="content-start"
    >
      <div
        class="page-header__full-width-container"
      >
        <div
          class="page-header__breadcrumbs"
        >
          <nav
            aria-label="Breadcrumbs"
            data-component="breadcrumbs"
            role="navigation"
          >
            <ol
              class="breadcrumbs"
            >
              <li
                class="breadcrumbs__crumb"
              >
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="breadcrumbs__crumb"
              >
                <a
                  href="/news"
                >
                  News
                </a>
              </li>
              <li
                class="breadcrumbs__crumb"
              >
                <span>
                  In-depth
                </span>
              </li>
            </ol>
          </nav>
          <script
            type="application/ld+json"
          >
            {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/",
        "name": "Home"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/news",
        "name": "News"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "name": "In-depth"
      }
    }
  ]
}
          </script>
        </div>
        <h1
          class="page-header__heading "
        >
          In-depth
        </h1>
        <p
          class="page-header__lead"
        >
          Our in-depth articles explore how our work is making a real difference to people's lives.
        </p>
        <div
          class="page-header__description"
        >
          Through this featured content, you'll learn how our guidance and advice is supporting effective decision making, for the benefit of individuals and society as a whole.
        </div>
      </div>
    </div>
    <nav
      aria-label="News section navigation"
      class="horizontal-nav"
    >
      <ul
        class="horizontal-nav__list"
      >
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/news"
          >
            News
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/news/articles"
          >
            News articles
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="true"
            class="horizontal-nav__link"
            href="/news/in-depth"
          >
            In-depth
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/news/blogs"
          >
            Blogs
          </a>
        </li>
        <li
          class="horizontal-nav__item"
          data-component="horizontal-nav"
        >
          <a
            aria-current="false"
            class="horizontal-nav__link"
            href="/news/podcasts"
          >
            Podcasts
          </a>
        </li>
      </ul>
    </nav>
    <h2
      class="visually-hidden"
      tabindex="-1"
    >
      In-depth articles list
    </h2>
    <ul
      class="newsList"
    >
      <li
        class="newsItem"
      >
        <article
          class="newsCard listItem"
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/news/articles/new-story"
            style="background-image: url(https://local-host-test-nice-org.com/f/292509/805x453/42b7a1a419/nhs-team-16-9.JPG/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/news/articles/new-story"
              >
                new story
              </a>
            </h3>
            <p>
              test
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <time
                  class="date"
                  datetime="2025-05-10 00:00"
                >
                  10 May 2025
                </time>
              </div>
            </footer>
          </div>
        </article>
      </li>
      <li
        class="newsItem"
      >
        <article
          class="newsCard listItem"
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/news/articles/young-people-with-chronic-tic-disorders-and-tourette-syndrome"
            style="background-image: url(https://local-host-test-nice-org.com/f/292509/x/43838fb421/storyblok_1.avif/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/news/articles/young-people-with-chronic-tic-disorders-and-tourette-syndrome"
              >
                Young people with chronic tic disorders and Tourette syndrome
              </a>
            </h3>
            <p>
              Digital therapy to help children and young people manage their condition.
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <time
                  class="date"
                  datetime="2025-05-09 00:00"
                >
                  09 May 2025
                </time>
              </div>
            </footer>
          </div>
        </article>
      </li>
      <li
        class="newsItem"
      >
        <article
          class="newsCard listItem"
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/news/articles/technology-for-children-and-young-people-with-chronic-tic-disorders-and-tourette-syndrome-recommended"
            style="background-image: url(https://local-host-test-nice-org.com/f/292509/x/43838fb421/storyblok_1.avif/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/news/articles/technology-for-children-and-young-people-with-chronic-tic-disorders-and-tourette-syndrome-recommended"
              >
                Technology for children and young people with chronic tic disorders and Tourette syndrome recommended
              </a>
            </h3>
            <p>
              Final guidance recommends one digital therapy to help children and young people manage their condition.
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <time
                  class="date"
                  datetime="2025-05-09 00:00"
                >
                  09 May 2025
                </time>
              </div>
            </footer>
          </div>
        </article>
      </li>
      <li
        class="newsItem"
      >
        <article
          class="newsCard listItem"
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/news/articles/test-news-article-8th-may"
            style="background-image: url(https://local-host-test-nice-org.com/f/292509/624x350/b118a9ebd8/0c34ac90-2a34-11f0-b26b-ab62c890638b-jpg.webp/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/news/articles/test-news-article-8th-may"
              >
                Test news article 8th may
              </a>
            </h3>
            <p>
              intro text test
Text update 09/05/25 10:15
Text update 09/05 10:25
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <time
                  class="date"
                  datetime="2025-05-08 00:00"
                >
                  08 May 2025
                </time>
              </div>
            </footer>
          </div>
        </article>
      </li>
      <li
        class="newsItem"
      >
        <article
          class="newsCard listItem"
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/news/articles/thousands-of-people-in-england-with-sight-loss-could-benefit-from-a-new-treatment-option"
            style="background-image: url(https://local-host-test-nice-org.com/f/243782/8660x5773/c394fd5edb/sight-test.jpg/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/news/articles/thousands-of-people-in-england-with-sight-loss-could-benefit-from-a-new-treatment-option"
              >
                Thousands of people in England with sight loss could benefit from a new treatment option
              </a>
            </h3>
            <p>
              Test text update 08/05/25 16:41
Test text update 08/05/25 16:45
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <time
                  class="date"
                  datetime="2025-05-08 00:00"
                >
                  08 May 2025
                </time>
              </div>
            </footer>
          </div>
        </article>
      </li>
      <li
        class="newsItem"
      >
        <article
          class="newsCard listItem"
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/news/articles/test-news-story-6-may-afternoon"
            style="background-image: url(https://local-host-test-nice-org.com/f/292509/624x350/b118a9ebd8/0c34ac90-2a34-11f0-b26b-ab62c890638b-jpg.webp/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/news/articles/test-news-story-6-may-afternoon"
              >
                Test news story 6 may afternoon
              </a>
            </h3>
            <p>
              Intro text
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <time
                  class="date"
                  datetime="2025-05-06 00:00"
                >
                  06 May 2025
                </time>
              </div>
            </footer>
          </div>
        </article>
      </li>
    </ul>
    <section
      class="action-banner"
      data-component="action-banner"
    >
      <div
        class="action-banner__container"
      >
        <div
          class="action-banner__inner"
        >
          <div
            class="action-banner__text"
          >
            <h2
              class="action-banner__title"
            >
              Sign up for our newsletters and alerts
            </h2>
            <div
              class="action-banner__intro"
            >
              Keeping you up to date with important developments at NICE.
            </div>
          </div>
          <div
            class="action-banner__actions"
          >
            <a
              class="btn btn--cta signupCta"
              data-component="button--cta"
              href="/nice-newsletters-and-alerts"
            >
              Sign up for newsletters and alerts
            </a>
          </div>
        </div>
      </div>
    </section>
    <div
      class="simple-pagination"
      data-component="simple-pagination"
    >
      <p>
        Page 
        <b>
          1
        </b>
         
        of 
        <b>
          2
        </b>
      </p>
      <nav
        aria-label="Pagination"
      >
        <span
          class="simple-pagination__link-wrapper"
        >
          <a
            class="simple-pagination__link"
            href="/news/in-depth?page=2"
          >
            Next page
          </a>
           
        </span>
      </nav>
    </div>
  </div>
</body>
`;

exports[`/news/in-depth/index.page InDepthArticleIndexPage should render error page if validateRouteParams returns error 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumbs"
    data-component="breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Error
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/",
        "name": "Home"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "name": "Error"
      }
    }
  ]
}
  </script>
  <div
    class="page-header  page-header--vertical-padding-loose"
    data-component="page-header"
    id="content-start"
  >
    <h1
      class="page-header__heading "
    >
      An error returned from getserversideprops
    </h1>
    <p
      class="page-header__lead"
    >
      We'll look into it right away. Please try again in a few minutes. And if it's still not fixed, 
      <a
        href="/get-involved/contact-us"
      >
        contact us
      </a>
      .
    </p>
  </div>
  <h2>
    What can I do now?
  </h2>
  <div
    class="grid grid--loose"
  >
    <div
      data-component="grid-item"
      data-g="12 sm:6 md:4"
    >
      <ul
        class="mt--0"
      >
        <li>
          Check that the web address has been typed correctly
        </li>
        <li>
          Look for it using search
        </li>
        <li>
          Browse for it from the 
          <a
            href="/"
          >
            homepage
          </a>
        </li>
      </ul>
      <p>
        And if you still can't find what you're looking for, 
        <a
          href="/get-involved/contact-us"
        >
          get in touch
        </a>
        .
      </p>
    </div>
    <div
      data-component="grid-item"
      data-g="12 sm:6 md:8"
    >
      <p
        class="mt--0"
      >
        Here's where you can go from here:
      </p>
      <ul>
        <li>
          Explore 
          <a
            href="/guidance"
          >
            NICE guidance, quality standards and advice
          </a>
          . You can do this by 
          <a
            href="/guidance/conditions-and-diseases"
          >
            category to find a topic
          </a>
          , or choose a 
          <a
            href="/guidance/published"
          >
            list organised by date or title
          </a>
          .
        </li>
        <li>
          <a
            href="/about"
          >
            Visit our about pages
          </a>
           to find information about our services, 
          <a
            href="/about/who-we-are"
          >
            who we are
          </a>
           and 
          <a
            href="/about/what-we-do"
          >
            what we do
          </a>
        </li>
      </ul>
      <p>
        If this is not the first time you have seen this message and believe that there is a serious problem 
        <a
          href="/get-involved/contact-us"
        >
          please contact us directly
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
